src/main/java/edu/kit/compiler/codegen/pattern/InstructionMatch.java

```diff
@@ -0,0 +1,150 @@
+package edu.kit.compiler.codegen.pattern;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import edu.kit.compiler.codegen.ExitCondition;
+import edu.kit.compiler.codegen.PhiInstruction;
+import edu.kit.compiler.intermediate_lang.Instruction;
+import firm.nodes.Node;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+
+public interface InstructionMatch extends Match {
```

Comment:
As this is an interface now, you can remove the `public`, `abstract` and `static`modifiers

Reply:
I definitely can't remove the `static` modifiers, but you're right about the other two.